packages/roosterjs-content-model/lib/editor/coreApi/createPasteModel.ts

```diff
@@ -0,0 +1,70 @@
+import ContentModelBeforePasteEvent from '../../publicTypes/event/ContentModelBeforePasteEvent';
+import domToContentModel from '../../domToModel/domToContentModel';
+import { ContentModelEditorCore, CreatePasteModel } from '../../publicTypes/ContentModelEditorCore';
+import {
+    createDefaultHtmlSanitizerOptions,
+    createFragmentFromClipboardData,
+    wrap,
+} from 'roosterjs-editor-dom';
+import { ClipboardData, EditorCore, PluginEventType, NodePosition } from 'roosterjs-editor-types';
+
+export const createPasteModel: CreatePasteModel = (
+    core: ContentModelEditorCore,
+    clipboardData: ClipboardData,
+    position: NodePosition | null,
+    pasteAsText: boolean,
+    applyCurrentStyle: boolean,
+    pasteAsImage: boolean = false
+) => {
+    if (!clipboardData) {
+        return null;
+    }
+
+    // Step 1: Prepare BeforePasteEvent object
+    const event = createBeforePasteEvent(core, clipboardData);
+
+    const fragment = createFragmentFromClipboardData(
+        core,
+        clipboardData,
+        position,
+        pasteAsText,
+        applyCurrentStyle,
+        pasteAsImage,
+        event
+    );
+
+    return domToContentModel(wrap(fragment, 'span'), core.api.createEditorContext(core), {
```

Comment:
is it necessary to call `wrap` here?

Reply:
Yes, domToContentModel's first parameter only accept HTMLElement, Document fragment seem to be not compatible. Wrapping it a span element should work.

Reply:
Can you try to make domToContentModel accept Node instead of HTMLElement, so no need to wrap here